Investigate esArchiver flakiness

pheyos opened the pull request:
DO NOT MERGE

This PR investigates a flaky esArchiver behavior when loading and unloading the same archive.

Related to the investigation done in #223043 (actually taking over that reproduction test case for further investigation)

pheyos committed:
Investigate esArchiver flakiness

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8707](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8707)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8710](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8710)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Increase test count to 400, temporarily remove index exist check

pheyos commented:
No failure in the first 10k runs. Increased the test count to 400 and temporarily removed the index exists check (to not modify the timing in case that's related).

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8716](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8716)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Running with basic license and ssl enabled

pheyos commented:
Noticed that the Security solution [config](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_api_integration/config/ess/config.base.basic.ts) used for the experiment in #223043 is running with basic license and ssl enabled - trying to reproduce with that setup.

pheyos committed:
Update license to trial

pheyos committed:
Switch to trial license (correct config this time)

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8722](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8722)
[❌] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 1/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Switch back to basic license, disable ssl

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8724](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8724)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos commented:
Summary of the recent experiment with license and ssl settings:

license \ ssl | ssl: true | ssl: false
--- | --- | ---
license: basic | :x: | :heavy_check_mark: 
license: trial | :x: | :heavy_check_mark: 

So it looks like changing the ES ssl flag to `true` for the server configuration causes this esArchiver flakiness to happen.

pheyos committed:
Switch back to trial license and ssl enabled

pheyos committed:
Add more logging to esArchiver index delete

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8727](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8727)
[❌] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 5/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Add more logging

pheyos committed:
More logging

pheyos committed:
Set test count back to 400

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8729](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8729)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
More logging on load

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8739](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8739)
[❌] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 24/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
More load logging

pheyos committed:
Set concurrency to 1

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8741](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8741)
[❌] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 0/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Enable bulk refreshOnCompletion

pheyos committed:
Check es bulk directly

pheyos commented:
The ES bulk operation within esArchiver is leading to the 2 ingested documents.
The second one only shows up after refreshing the index.
Testing the ES bulk upload setup outside of esArchiver next.

pheyos committed:
Set bulk retries to 1

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8746](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8746)
[❌] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 22/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Check es non-helper bulk

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8747](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8747)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

bartoval committed:
[ES|QL] fix: enable partial license check for suggestions and fix license position in signatures (#228965)

## Summary

This PR is part of https://github.com/elastic/kibana/issues/216791 and
is a blocker for other pending PRs

- Added license to autogenerated signatures
- Enable hint filtering for licensed function inputs (example
ST_EXTENT_AGG)
- Added a complete test
-
example:
Basic license hide functions with license

1) hide CATEGORIZE (no partial license)
<img width="1330" height="187" alt="basic_categorize"
src="https://github.com/user-attachments/assets/ab2ca2ba-5546-4b8a-bb2a-a48e4bfbbce7"
/>

2) show ST_EXENT_AGG, but not licensed inputs lilke TO_CARTESIANSHAPE()
<img width="1283" height="298" alt="basic_st_extent_agg"
src="https://github.com/user-attachments/assets/449a6987-933a-4572-83d7-ef0cba0f22ac"
/>


Then the Trial license show all
<img width="1336" height="138" alt="trial_categorize"
src="https://github.com/user-attachments/assets/75886bf1-ddf7-4916-a5a8-bd57a92f73a6"
/>

<img width="1589" height="328" alt="trial_st_extent_agg"
src="https://github.com/user-attachments/assets/c8b21289-6358-40b3-b2f2-331f597803de"
/>



to test this feat:

start with trial license and verify
change license to basic ( From the sidebar -> stack management ->
license management -> then change to basic)
verify with basic license

tsullivan committed:
docs: Improve documentation for navigation services and components (#229034)

## Summary

This PR adds comprehensive documentation for Kibana's navigation system
across multiple components:

1. ProjectNavigationService:
- Creates detailed
`src/core/packages/chrome/browser-internal/src/project_navigation/README.md`
explaining architecture, data flow, and core functions
- Documents how navigation trees are registered in both stateful and
serverless deployments
   - Explains breadcrumbs generation and active item highlighting
   - Adds data flow visualization showing reactive streams

2. Navigation Component:
- Updates
`src/platform/packages/shared/shared-ux/chrome/navigation/README.mdx`
with architecture and component relationships
- Clarifies the roles of `NavigationKibanaProvider` and
`NavigationProvider`
   - Adds cross-references to related documentation

3. Navigation Plugin:
- Adds `src/platform/plugins/shared/navigation/public/README.md`,
explaining integrations with core services, gives component descriptions
and usage examples
     - include comprehensive information about the start contract
   - Adds `src/platform/plugins/shared/navigation/common/README.md`
   - Adds `/src/platform/plugins/shared/navigation/server/README.md`

4. Serverless Navigation Plugin:
- Documents serverless-specific navigation implementation in
`x-pack/platform/plugins/shared/serverless/public/navigation/README.md`
   - Clarifies differences from stateful deployment navigation
   - include comprehensive information about the start contract

5. Function-Level Documentation:
   - Adds detailed JSDoc comments to `useNavigation` hooks
- Distinguishes between hooks with the same name in different contexts

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

lcawl committed:
Update publish_oas_docs.sh for 9x-unreleased branch (#229197)

cauemarcondes committed:
[APM] Fix service error table columns sorting (#229199)

closes https://github.com/elastic/kibana/issues/229192


https://github.com/user-attachments/assets/ea78301b-7636-48cf-a048-89a08844de7f

hop-dev committed:
[Entity Analytics][Privmon] Deduplicate privileged users from index sync and CSV upload (#229179)

## Summary

Deduplicate entities in the same batch with the same user.name so that
only one will ever be created during index sync and csv upload.

## CSV upload test

Upload 1k_tiagos.csv, only one tiago should be added. (or 1k_marks.csv
to test a diff name)



[1k_tiagos.csv](https://github.com/user-attachments/files/21390988/1k_tiagos.csv)

[1k_marks.csv](https://github.com/user-attachments/files/21391494/1k_marks.csv)


## Index Sync Test
```
# create index with correct mapping
PUT /my-custom-index
{
  "mappings" : {
    "properties": {
      "user.name": {
        "type": "keyword"
      }
    }
  }
}

# mash cmd + enter until there are a few hundred tiagos
POST /_bulk
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }
{ "index": { "_index": "my-custom-index" } }
{ "user": { "name": "tiago" } }

# check how many tiagos you have
GET /my-custom-index/_search?size=0

# now go and add this index as an index sync source, there should only ever be one tiago: 
GET /.entity_analytics.monitoring.users-default/_search
```

---------

Co-authored-by: Tiago Vila Verde <tiago.vilaverde@elastic.co>
Co-authored-by: CAWilson94 <charlotte.wilson@elastic.co>

drewdaemon committed:
[ES|QL] support expressions in `SORT` (#223176)

## Summary

Part of https://github.com/elastic/kibana/issues/216492

**expressions in SORT!**


https://github.com/user-attachments/assets/83419eca-aa9b-4bfe-900c-e620fff12269


**puts the commas in the right place**


https://github.com/user-attachments/assets/572ba88d-1157-401f-b5f6-b50811739345

**handles ambiguity around `NULL` keyword**


https://github.com/user-attachments/assets/0cc332e1-e667-4874-9354-ad5c3788b180

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

animehart committed:
Fix for flickering page when opening findings flyout (#229016)

## Summary

This PR addresses the issue where Findings Flyout flickers when it first
being rendered


https://github.com/user-attachments/assets/13ef6536-3d8a-46ae-a551-6e74ec7c4f42

semd committed:
[Security Solution][Entity Analytics] Change hide timeline approach in priv mon onboarding (#229096)

## Summary

Remove the hook to update link config, in favour of the CSS approach.

### Why
As discussed with @michaelolo24 and @kqualters-elastic, the
`useUpdateLinkConfig` is too generic and too powerful. Every time the
function is called, the Security Solution plugin registry updater is
called to change the `deepLinks`, and it re-mounts the app.
It's very prone to misuse, and we don't want to open that door,
especially if it's created only to solve this small visual issue with
the timeline.

### Alternative
The "patchy but simple" approach proposed by @kqualters-elastic, which
hides the timeline updating the `element.style.display`, has been
applied. A `setTimeout` has been introduced to solve the problem of the
timeline not being rendered yet when the effect is executed.

@machadoum Please let me know what you think.

## Demo


https://github.com/user-attachments/assets/0347fddb-07bd-43f8-8899-6a10a9b2efd0

(The update of the status from "onboarding" to "dashboard" has been
artificially simulated for this demo.)

michaelolo24 committed:
[Security Solution][Response Ops] - Disable cases analytics index (#229204)

## Summary

Disabling cases analytics index

jughosta committed:
[OneDiscover] Make sure that contextual profiles have unique and valid ids (#229167)

- Closes https://github.com/elastic/kibana/issues/204253

## Summary

This PR adds tests to validate that the defined profile ids are unique
and correspond to the specified context level.


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

jughosta committed:
[Discover] Update the logic of selecting a sub field to be used when fetching field stats (#228969)

- Fixes https://github.com/elastic/kibana/issues/228952

## Summary

This PR fixes what field is being picked for fetching stats in the
sidebar popover. Previously, it was choosing one of the multi fields,
now it will choose the parent field first (sidebar groups similar fields
and displays them inside the popover).

### Checklist


- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

kibanamachine committed:
[api-docs] 2025-07-24 Daily api_docs build (#229234)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1143

tomsonpl committed:
[EDR Workflows] Show error when fetching scripts fails (#226532)

mohamedhamed-ahmed committed:
[Dataset Quality] Fix failing tests (#229104)

fixes https://github.com/elastic/kibana/issues/222500

ersin-erdal committed:
Fix flaky action error test (#229138)

Resolves: #209913 

The test case was finding the rule from the previous test case because
it was running before the objectRemover removes all the savedObjects,
due to lack of an "await".

alexwizp committed:
Auto-fix violations of the @elastic/eui/require-aria-label-for-modals ESLint rule (#227623)

> [!CAUTION]
> ⚠️ Changes were made by GenAI. Please review carefully.

> [!NOTE]
> Only `EuiConfirmModal` part

## Summary
This PR applies the auto-fix for the newly introduced
`@elastic/eui/require-aria-label-for-modals` Ensures that EUI modal
components `EuiConfirmModal` have either an `aria-label` or
`aria-labelledby` prop for accessibility. This helps screen reader users
understand the purpose and content of modal dialogs.

## Changes

- For **function components**:
1. Verify whether `useGeneratedHtmlId` is imported from the
`@elastic/eui` package. If not, add the following import.
2. Inside the render method, define a new constant: `const
confirmModalTitleId = useGeneratedHtmlId();`
- For **class components**:
1. Verify whether `htmlIdGenerator` is imported from the `@elastic/eui`
package. If not, add the following import.
2. Inside the render method, define a new constant: `const
confirmModalTitleId = htmlIdGenerator()('confirmModalTitle');`

In both cases, add the following props to the `EuiConfirmModal`
component:
- `aria-labelledby={confirmModalTitleId}`
- `titleProps={{ id: confirmModalTitleId }}`

## Related
- elastic/eui#8843

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

e40pud committed:
[Attack Discovery][Scheduling] Add integration tests to cover AD scheduling rule management endpoints (#12400) (#229076)

## Summary

Main ticket ([Internal
link](https://github.com/elastic/security-team/issues/12400))

This is a followup to https://github.com/elastic/kibana/pull/228061
where basic integration tests were added. This PR focuses on RBAC and
CRUD attack discovery schedules in different spaces.

This PR adds more integration tests for the attack discovery schedule
feature API routes:
- Create AD scheduling rule: `POST
/internal/elastic_assistant/attack_discovery/schedules`
- Read AD scheduling rule by id: `GET
/internal/elastic_assistant/attack_discovery/schedules/{id}`
- Update AD scheduling rule by id: `PUT
/internal/elastic_assistant/attack_discovery/schedules/{id}`
- Enable AD scheduling rule by id: `POST
/internal/elastic_assistant/attack_discovery/schedules/{id}/_enable`
- Disable AD scheduling rule by id: `POST
/internal/elastic_assistant/attack_discovery/schedules/{id}/_disable`
- Delete AD scheduling rule by id: `DELETE
/internal/elastic_assistant/attack_discovery/schedules/{id}`
- Find all existing AD scheduling rules: `GET
/internal/elastic_assistant/attack_discovery/schedules/_find`

### Flaky Test Runner

* **Basic/Essentials**: [100 ESS + 100
Serverless](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8793)
* **Trial/Complete**: [100 ESS + 100
Serverless](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8796)

pheyos committed:
FTR - Update ML DFA settings in serverless configs (#228908)

## Summary

This PR updates all the serverless FTR config files to no longer disable
machine learning data frame analytics ( `xpack.ml.dfa.enabled`) after
this has recently been enabled for Observability projects on the
Elasticsearch side.

As part of this change, the PR also removes a few outdated
`xpack.ml.ad.enabled` entries.

maximpn committed:
[Security Solution] Implement prebuilt rule upgrade notifications test plan (#225912)

**Addresses:** https://github.com/elastic/kibana/issues/202078

## Summary

This PR implements the [Prebuilt Rules Upgrade Notifications test plan](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_upgrade_notifications.md).

## Details

This PR implements test scenarios listed in the [Prebuilt Rules Upgrade Notifications test plan](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_upgrade_notifications.md). Unit tests listed in the test plan was skipped due to low testability of the UI code, high maintenance effort and low total value since the same scenarios are covered by the e2e tests.

## Flaky test runner

- ✅ [added integration tests (200 runs)](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8656)
- ✅ [added e2e tests (75 runs)](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8664) Since the whole Rule Management Cypress tests group was ran you may observe some failures in unrelated to this PR tests.

bartoval committed:
[ES|QL]: Integrate Elasticsearch Command Metadata into ESQL Registry (#229172)

## Summary

This PR is part of https://github.com/elastic/kibana/issues/216791 and a
prequel of https://github.com/elastic/kibana/pull/228897

- new script to generate metadata for commands, from elasticsearch
- compatibility script to generate functions and commands using the
usual syntax yarn make:defs <path of elasticsearch>
- adds information to the command registry
- Test the logic

<img width="1325" height="572" alt="pr"
src="https://github.com/user-attachments/assets/fae46153-c294-421b-9c4c-aa105de4bb9f"
/>

rStelmach committed:
[One Discover] Prevent search highlighting markup in logs overview field action filters (#227652)

closes : #226375
## Summary 📚 
Fixes an issue where HTML search highlighting markup (`<mark>` tags) was
incorrectly carried over into filter values when using field actions
"Filter for value" in the logs overview tab.

### Problem 🐛 
When searching in Discover and using field actions in the logs overview
tab's content breakdown section, the generated filters would contain
HTML markup instead of clean field values.
The logs overview components were using the formatted/highlighted field
values (which contain ` <mark> ` tags for visual highlighting) directly
in field actions, instead of accessing the original, clean field values
from the raw document.

### Solution
Instead of trying to strip HTML markup (which could accidentally remove
legitimate HTML content from log messages), this fix implements a
cleaner approach which is passing raw document data.

## Demo 🎥 


https://github.com/user-attachments/assets/286db889-e54b-4026-b5bc-0e3ca9e1ea58

## How to test 🔬 

- Ingest log data (e.g by using Synthtrace node scripts/synthtrace.js
simple_logs)
- Write a search query that matches part or all of the message
- Open the doc flyout and use an action for the message field

## Open discussion 🗣️ 
This solution is also handling an edge case when a message can contain
`<mark>` html element and that's why we are passing raw doc message.
If that is too much we can just use regex to filter out `<mark>` html
markups caused by our highlighting. LMK

tonyghiani committed:
[Streams] Udpate processors doc links (#229262)

## 📓 Summary

Closes #229250 

After being refactored with
https://github.com/elastic/kibana/pull/219286, the new doc links for
processors are accessible through the core docLink map. This work fixes
the processors' broken links.

jloleysens committed:
[OAS] Validation check (#228402)

## Summary

A CLI tool that can help devs while they develop their OAS and,
ultimately be used tp keep track of the increase/decrease in issues in
our generated OAS.

This is a simple first iteration that developers can use to check
manually. Next steps would be to integrate this into PR checks.

Help:

```
❯ node ./scripts/validate_oas_docs.js --help

  node ./scripts/validate_oas_docs.js

  Validate Kibana OAS YAML files (in oas_docs/output)

  Options:
    --path             Pass in the (start of) a custom path to focus OAS validation error reporting, can be specified multiple times.
    --only             Validate only OAS for the a specific offering, one of "traditional" or "serverless". Omitting this will validate all offerings.
    --verbose, -v      Log verbosely
    --debug            Log debug messages (less than verbose)
    --quiet            Only log errors
    --silent           Don't log anything
    --help             Show this message

  Examples:
    node ./scripts/validate_oas_docs.js
    node ./scripts/validate_oas_docs.js --path /paths/~1api~1fleet~1agent_policies --path /paths/~1api~1fleet~1agent_policies
    node ./scripts/validate_oas_docs.js --only serverless --path /paths/~1api~1fleet~1agent_policies
```

## Uses 3rd party dep for validation

I am open to avoiding the use of `@seriousme/openapi-schema-validator`
since it is a thin wrapper around `ajv`, but it's nice to not have the
additional YAML files lying around for this CLI. I am basically trying
to simulate the feedback you might get from swagger editor. One
alternative is that we run this check with puppeteer and use the actual
Editor UI ([GH
action](https://github.com/swaggerexpert/swagger-editor-validate)), but
this seemed like a simpler place to start.

**Purpose**
Used to validate OAS 3 that Kibana generates (bundles AJV and some OAS 3
schema descriptions for convenience) at _dev time only_

**Justification**
We can avoid placing large OAS 3 schemas in our own code

**Alternatives explored**
There are several other OAS 3 validation libraries, but none that worked
quite as simply as this one. Would be pretty straight-forward to remove
it in future (or now) if we feel strongly about avoiding it.

**Existing dependencies**
AJV, but AJV is not specialised to OAS 3 validation

Outputs like:

<details>

<summary>A wall of text</summary>

```
 info About to validate spec at /Users/jeanlouisleysens/repos/work/kibana/oas_docs/output/kibana.serverless.yaml
   │ warn /Users/jeanlouisleysens/repos/work/kibana/oas_docs/output/kibana.serverless.yaml is NOT valid
   │ warn Found the following issues
   │
   │      /paths/~1api~1fleet~1agent_policies/get/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies/get/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies/post/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies/post/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1_bulk_get/post/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1_bulk_get/post/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}/get/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}/get/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}/put/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}/put/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1auto_upgrade_agents_status/get/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1auto_upgrade_agents_status/get/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1copy/post/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1copy/post/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1download/get/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1download/get/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1download/get/responses/404
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1full/get/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1full/get/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1outputs/get/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1{agentPolicyId}~1outputs/get/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1delete/post/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1delete/post/responses/400
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1outputs/post/responses/200
   │      must have required property 'description'
   │
   │      /paths/~1api~1fleet~1agent_policies~1outputs/post/responses/400
   │      must have required property 'description'
   │
   │ warn Found 25 errors in /Users/jeanlouisleysens/repos/work/kibana/oas_docs/output/kibana.serverless.yaml
 info Done
 ```
 
 </details>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jgowdyelastic committed:
[ML] Disabling part of the forecast test (#226209)

Disabling check for the forecast section of the single metric chart as a
[race condition](https://github.com/elastic/kibana/issues/229143) in how
the appState is loaded can sometimes mean the slider does not
automatically move to the forecast section of the chart.

Fixes #164381

KDKHD committed:
[Security Solution] [AI assistant] Give Security AI assistant context about the signal index outside of the security solution (#229173)

## Summary

Summarize your PR. If it involves visual changes include a screenshot or
gif.

There is a bug where the Alers Count tool is unavailable when using the
Security AI assistant outside of the security solution (Global Security
AI assistant). This PR partially fixes this issue.

### How to test
- Start Kibana
- Head to stack management > Advanced settings > Assistant visibility
and set it to "Security AI assistant in other solutions"
- Visit the security solution (e.g. Security Alerts)
- Vist the Discover app
- Ask the assistant how many alerts you have open. Expect an answer.
Previously, the alerts counts tool was only available when invoking the
assistant from the security solution.

It is important to note that the signal index (which is required by the
alerts count tool) will only become available to the assistant after
visiting the security solution first. After visiting the security
solution, you can go to another solution and use the alerts count tool.

Ideally, the user would not need to visit the security solution first,
however, the signal index is fetched in
[this](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_signal_index.tsx)
hook and some more refactoring would be required to move some of this
logic into plugin setup (and I think there is some additional complexity
with do this).

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [X] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [X]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [X] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [X] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [X] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [X] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [X] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [X] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

machadoum committed:
[SecuritySolution] Add FTR test for privmon label upload via CSV (#229272)

## Summary

Add FTR test for privmon label upload via CSV

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

dgieselaar committed:
[Inference] Evaluations framework (#227453)

introduces `@kbn/evals` to enable offline evaluation suites for
LLM-based workflows in Kibana, using:

- `@kbn/scout` and Playwright for running the evaluations
- `@kbn/inference-plugin` for handling LLM interactions
- [Phoenix](https://github.com/Arize-ai/phoenix) for storing evaluations

See:
https://github.com/dgieselaar/kibana/blob/evaluation-runner/x-pack/platform/packages/shared/kbn-evals/README.md

Other changes in this PR:

- Implement support for `exports` in package.json in
`@kbn/import-resolver` (this causes some eslint-disables to no longer be
necessary)
- Graceful shutdown for OpenTelemetry span processors
- Add `scripts/playwright.js` script which sets up babel transforms,
tracing etc

Notes:
- used o3 to implement support for `exports` (and then iterated on it)
- used o3 to generate a
[README](https://github.com/dgieselaar/kibana/blob/evaluation-runner/x-pack/platform/packages/shared/kbn-evals/README.md)
(and made some minor changes)

Dependencies:
- Added @arizeai/phoenix-client to interact with Phoenix's API. The
alternative is using REST/GraphQL directly but there's a lot of
client-side logic in the SDK.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Tiago Costa <tiago.costa@elastic.co>
Co-authored-by: Tiago Costa <tiagoffcc@hotmail.com>
Co-authored-by: Viduni Wickramarachchi <viduni.ushanka@gmail.com>

flash1293 committed:
🌊 Streams: Show full error message on state changes (#229148)

We didn't do so so far because theoretically there can be multiple
issues on changing streams state, so this is an aggregate error, with
the top level message being useless.

## Before

<img width="352" height="167" alt="Screenshot 2025-07-23 at 15 54 26"
src="https://github.com/user-attachments/assets/5d59e7bc-d79f-46e2-964a-9113644632a5"
/>

## After

<img width="367" height="234" alt="Screenshot 2025-07-23 at 15 53 47"
src="https://github.com/user-attachments/assets/21837365-3c29-45bc-87f4-7b25d5f90257"
/>

kibanamachine committed:
skip failing test suite (#176027)

rmyz committed:
[Profiling] Fix Differential Functions Cypress Test (#229276)

k-g-elastic committed:
Security quality gate Explore team notify changes (#229219)

The former Security Explore team responsibilities have been distributed
across a few other teams in Security (Entity Analytics, Investigations,
Cases). This PR reassigns the Cypress and FTR tests to the new owners
for the pipelines.

jedrazb committed:
[onechat] Improve user-facing error messages in toasts (#229254)

## Summary

- Add onechat-common method to extract actual error msg from the onechat
error object
- That impl was inspration
https://github.com/elastic/kibana/blob/main/src/core/packages/fatal-errors/browser-internal/src/get_error_info.ts#L15
(but it's private + contains some cases we don't care about)
- All onechat errors are instances of
[ServerSentEventError](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-sse-utils/src/errors.ts#L18)
and the default `err.message` property is not populated
- Update toasts 

Verified locally that now `Bad Request` text is actually changed with
the error body.message

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

ymao1 committed:
Fixing flaky `alert deletion` tests (#228874)

Fixes https://github.com/elastic/kibana/issues/227349
Fixes https://github.com/elastic/kibana/issues/227523
Fixes https://github.com/elastic/kibana/issues/228753
Fixes https://github.com/elastic/kibana/issues/228584
Fixes https://github.com/elastic/kibana/issues/228341
Fixes https://github.com/elastic/kibana/issues/227844
Fixes https://github.com/elastic/kibana/issues/227808
Fixes https://github.com/elastic/kibana/issues/228821


## Summary

These tests haven't been skipped, just trying to pre-emptively address.
Changing to find the latest event log doc written ordered by descending
timestamp.

Flaky test runner runs:
*
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8781
*
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8774

kgeller committed:
[AI4DSOC] Updating integrations skeleton now that we have 6 integration tiles (#228470)

## Summary

With the addition of the Elastic Security integration, we now have 6
integrations, I've updated the loading skeleton to have 6 tiles as well.

## Screenshots

Loading:
Before:
<img width="1728" height="1117" alt="Screenshot 2025-07-17 at 10 25
04 AM"
src="https://github.com/user-attachments/assets/4b097d4e-6619-4c45-8897-dbc7527c0888"
/>

After:
<img width="1728" height="1117" alt="Screenshot 2025-07-17 at 10 26
20 AM"
src="https://github.com/user-attachments/assets/8ac89f5f-1dce-4a34-a446-5510489c938b"
/>

Loaded:
<img width="1728" height="1117" alt="Screenshot 2025-07-17 at 10 26
28 AM"
src="https://github.com/user-attachments/assets/3475d815-f422-4a9e-b401-1a758d9a0298"
/>

kibanamachine committed:
skip failing test suite (#228131)

alexwizp committed:
Auto-fix violations of the @elastic/eui/require-aria-label-for-modals ESLint rule (Step 2) (#227821)

> \[!CAUTION]
> ⚠️ Changes were made by GenAI. Please review carefully.

## Summary

This PR applies the auto-fix for the newly introduced
`@elastic/eui/require-aria-label-for-modals` rule. This rule ensures
that EUI modal components (`EuiModal` and `EuiFlyout`) include either an
`aria-label` or `aria-labelledby` prop to support screen reader
accessibility. These attributes help users understand the purpose and
content of modal dialogs.

## Changes Made

1. **Identified the modal title/header element** used within `EuiModal`
or `EuiFlyout`.
Common elements include: `EuiFlyoutTitle`, `EuiModalTitle`, `EuiTitle`,
`<h2>`, `<h3>`, or other heading tags.

2. **Ensured the title element has an `id`:**
If the header element lacked an `id`, one was programmatically
generated:

   * **For function components:**

* Imported `useGeneratedHtmlId` from `@elastic/eui` if not already
present:

       ```ts
       import { useGeneratedHtmlId } from '@elastic/eui';
       ```
     * Defined a unique ID before the `return` statement:

       ```ts
       const modalTitleId = useGeneratedHtmlId();
       ```
     * Applied the ID to the title element:

       ```tsx
       <EuiModalTitle id={modalTitleId}>Title</EuiModalTitle>
       ```

   * **For class components:**

* Imported `htmlIdGenerator` from `@elastic/eui` if not already present:

       ```ts
       import { htmlIdGenerator } from '@elastic/eui';
       ```
     * Defined a unique ID within the `render()` method:

       ```ts
       const modalTitleId = htmlIdGenerator()('modalTitle');
       ```
     * Applied the ID to the title element:

       ```tsx
       <EuiModalTitle id={modalTitleId}>Title</EuiModalTitle>
       ```

3. **Updated the `EuiModal` or `EuiFlyout` component** to include the
`aria-labelledby` attribute referencing the generated title ID:

   ```tsx
   <EuiModal aria-labelledby={modalTitleId} />
   ```

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

TattdCodeMonkey committed:
[Search][Tests] refactor - move stateful playground tests (#229213)

## Summary

Moving the playground stateful test under the function_search folder to
share configs. This is in preperation for enabling saved playground
which is already in `functional_search` and will be moved into the same
search_playground index when the feature is ungated.

mykolaharmash committed:
[Oblt Onboarding] Switch K8S Elastic Agent flow to Helm Chart (#227037)

Closes https://github.com/elastic/observability-dev/issues/4616

**🚨 Do not merge until Elastic Agent 9.0.4 is released 🚨**

This change replaces Kustomize in the code snippet for K8S (Elastic
Agent) onboarding flow with Helm Chart, which has support for disabling
metrics collection needed for Logs Essentials.

![CleanShot 2025-07-08 at 14 54
12@2x](https://github.com/user-attachments/assets/dbea746b-5135-4a74-ab29-49a9824f0523)

## How to test

There are two Kibana instances deployed from this PR: complete
serverless (see latest mention from kibanamachine in the comments) and
logs essentials (see "Observability Deployment" annotation on the
Buildkite job), this change needs to be tested on both.

1. Create a local k8s cluster (you can use [a reference stack
repo](https://github.com/elastic/oblt-reference-stack) for example,
follow instructions in the README)
2.  Use the Kibana complete serverless instance deployed from this PR
3. Go to Add Data  → Kubernetes → Elastic Agent onboarding flow
4. Run the command snippet in your terminal
5. Wait until data arrives and make sure metrics dashboard shows the
data
6. Now go through the same process on logs essentials instance and make
sure that only logs are ingested (the metrics dashboard should stay
empty but logs should be visible in Discover)

---------

Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>

kibanamachine committed:
skip failing test suite (#228130)

kdelemme committed:
chore(streams): redirect sig events to Discover ESQL (#228803)

pheyos committed:
bulk helper with 0 retries

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8833](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8833)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos commented:
Update: The flakiness also reproduced with `retries: 0`.

pheyos committed:
Merge branch 'main' into invetigate_es_archiver_flake

pheyos committed:
Ingest with fixed id

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8849](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8849)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
esArchiver test with generated id

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8856](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8856)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos commented:
With setting a fixed doc id in my es bulk helper experiment, the flakiness didn't show up in 20k tests anymore.

Then applied this to esArchiver: generating a uuid if the doc doesn't come with an id from the archive already. And this also didn't reproduce the flakiness in 30k tests anymore.

This behavior is matching what the [bulk docs](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-bulk) say:
> A create action fails if a document with the same ID already exists in the target An index action adds or replaces a document as necessary.

pheyos committed:
handle data streams

pheyos committed:
Don't generate id and don't retry on transport level

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8863](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8863)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8866](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8866)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

pheyos committed:
Check with generated id but excluded time series

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8869](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8869)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8870](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8870)
[✅] x-pack/platform/test/api_integration/apis/es_archiver/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228556%2Fhead)
